### PR TITLE
Use enviornment variables for watcher cli args

### PR DESCRIPTION
**Pull request by drGrove**
Using enviornment variable substitution for default cli args allows for modification of values via strategic merge patches instead of JSON6902. This reduces complexity of patches, allows for easier namespace overrides as well as a cleaner overlay

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [Tested your changes locally](https://github.com/tektoncd/results/blob/main/docs/DEVELOPMENT.md) (if this is a code change)
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user-facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contain the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

### CODE REVIEW

File: config/base/watcher.yaml [hunk 1]
```diff
@@ -36,12 +36,10 @@ spec:
...
           image: ko://github.com/tektoncd/results/cmd/watcher
           args:
-            [
-              "-api_addr",
-              "tekton-results-api-service.tekton-pipelines.svc.cluster.local:8080",
-              "-auth_mode",
-              "token",
-            ]
+            - -api_addr
+            - $(TEKTON_RESULTS_API_SERVICE)
```
Comment (james-callahan) [inline config/base/watcher.yaml:40]: Would this be cleaner?
```suggestion
            - -api_addr=$(TEKTON_RESULTS_API_SERVICE)
```

**Review by james-callahan — APPROVED** [review]

File: config/base/watcher.yaml
Comment (sayan-biswas) [inline config/base/watcher.yaml:1]: There are other config flags available for the watcher. So a long term solution would be to implement the knative configMap loader. Which will also reduce the complexity of patching. 

But I guess this okay for now.

File: config/base/watcher.yaml (hunk 1 above)
Comment (avinal) [reply config/base/watcher.yaml:40]: Either way seems fine.

**Review by avinal — APPROVED** [review]
Looks good to me
/approve

**Review by khrm — COMMENTED** [review]
/lgtm

**Pull request merged**